### ISSUE TITLE
Better way to set column values in listings

### ISSUE THREAD

**Issue by ramonski** [senaite/senaite.core#910]
## Steps to reproduce

1. Update the code to 1.2.8 and run the upgrade step
2. Navigate to /clients

## Current behavior

Traceback occurs

## Expected behavior

Client get listed

## Traceback

```
Traceback (innermost last):
  Module ZPublisher.Publish, line 138, in publish
  Module ZPublisher.mapply, line 77, in mapply
  Module ZPublisher.Publish, line 48, in call_object
  Module bika.lims.browser.bika_listing, line 581, in __call__
  Module Products.Five.browser.pagetemplatefile, line 125, in __call__
  Module Products.Five.browser.pagetemplatefile, line 59, in __call__
  Module zope.pagetemplate.pagetemplate, line 132, in pt_render
  Module five.pt.engine, line 98, in __call__
  Module z3c.pt.pagetemplate, line 158, in render
  Module chameleon.zpt.template, line 297, in render
  Module chameleon.template, line 191, in render
  Module chameleon.template, line 171, in render
  Module 0f7044e1c3bc14e5f7e029273927bd2d.py, line 525, in render
  Module eeb2449285a06966d652dfe0b11a8f70.py, line 1450, in render_master
  Module eeb2449285a06966d652dfe0b11a8f70.py, line 604, in render_content
  Module 0f7044e1c3bc14e5f7e029273927bd2d.py, line 446, in __fill_content_core
  Module five.pt.expressions, line 161, in __call__
  Module bika.lims.browser.bika_listing, line 1610, in contents_table
  Module bika.lims.browser.bika_listing, line 1788, in __init__
  Module bika.lims.browser.bika_listing, line 911, in folderitems
  Module bika.lims, line 217, in wrapper
  Module bika.lims.browser.bika_listing, line 1598, in _folderitems
  Module bika.lims.browser.clientfolder, line 159, in folderitem
KeyError: 'getClientID'

 - Expression: "provider:plone.abovecontentbody"
 - Filename:   ... rc/senaite/lims/skins/senaite_templates/main_template.pt
 - Location:   (line 146: col 90)
 - Source:     ... ent="structure provider:plone.abovecontentbody" tal:conditio ...
                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 - Expression: "here/main_template/macros/master"
 - Filename:   ... senaite.core/bika/lims/browser/templates/bika_listing.pt
 - Location:   (line 5: col 23)
 - Source:     metal:use-macro="here/main_template/macros/master"
                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 - Arguments:  repeat: {...} (0)
               template: <ViewPageTemplateFile - at 0x11607e550>
               views: <ViewMapper - at 0x119a36110>
               modules: <instance - at 0x10625bab8>
               args: <tuple - at 0x124f58d90>
               here: <ImplicitAcquisitionWrapper clients at 0x119c01910>
               user: <ImplicitAcquisitionWrapper - at 0x119c019b0>
               nothing: <NoneType - at 0x1050ace08>
               translate: <function translate at 0x1178535f0>
               container: <ImplicitAcquisitionWrapper clients at 0x119c01910>
               root: <ImplicitAcquisitionWrapper Zope at 0x11b093050>
               request: <instance - at 0x11c0ad908>
               wrapped_repeat: <SafeMapping - at 0x117b14db8>
               traverse_subpath: <list - at 0x11c20cb48>
               default: <object - at 0x105191bf0>
               loop: {...} (2)
               context: <ImplicitAcquisitionWrapper clients at 0x119c01910>
               view: <ClientFolderContentsView base_view at 0x11648de50>
               target_language: <NoneType - at 0x1050ace08>
               macroname: master
               options: {...} (0)
```

**Comment by ramonski** [comment]
Ok, the columns definition updated for clients and I was overriding these...

Maybe we should think about having a explicit `setter` for the column values that handles such cases better than just expecting the key to be defined and setting it:
`item["replace"]["getClientID"] = get_link(link_url, item["getClientID"])`

E.g. better would be like this:

```python
client_link = get_link(link_url, item.get("getClientID"))`
self.set_item("getClientID", client_link, replace=True)
```
and give some error logging back when it is not set.

**Comment by ramonski** [comment]
Will be resolved in Version 1.3 with the new ReactJS listings
